e2e/e2e.sh

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+set -e
+function bail {
+	echo "$*"
+	exit 1
+}
+
+function cleanup_server {
+	echo "The Server Log follows:"
+	cat "build/server_log.txt"
+	# Killing 0 sends the signal to all processes in the current
+	# process group
+	kill $SERVER_PID
+}
+
+# Travis CI is super cool but also uses ancient OS images
+if [ -f "/usr/bin/python3.6" ]; then
+	export PYTHON_BINARY="/usr/bin/python3.6"
```

Comment:
Why does this check specifically for the python 3.6 executable, and not just python3? On a normal setup python3 should point to the newest python3 available and python3 versions should be backwards compatible with other python3 versions.

Reply:
Because Travis CI runs Ubuntu 14.04 and the system Python can't cope with the typing package. So the `.travis.yml` installs python3.6 from the deadsnakes but that doesn't override the system python.